scripts/Muon/GUI/Common/contexts/phase_table_context.py

```diff
@@ -23,18 +23,18 @@ def add_phase_table(self, workspace):
     def get_phase_table_list(self, instrument):
         return [phase_table.workspace_name for phase_table in self.phase_tables if instrument in phase_table.workspace_name]
 
-    def add_phase_quad(self, workspace):
-        self.phase_quad = [item for item in self.phase_quad if item.workspace_name != workspace.workspace_name]
-        self.phase_quad.append(workspace)
+    def add_phase_quad(self, workspace, run_list):
+        self.phase_quad = {key: item for key, item in self.phase_quad.items() if item.workspace_name != workspace.workspace_name}
+        self.phase_quad.update({run_list: workspace})
```

Comment:
I think this line will overwrite the change from the previous one https://www.programiz.com/python-programming/methods/dictionary/update

Reply:
Good point, I think the update will do what is required here by itself.